sos/plugins/vdsm.py

```diff
@@ -0,0 +1,153 @@
+LVM_CONFIG = """
+global {
+    locking_type=0
+    use_lvmetad=0
+}
+devices {
+    preferred_names=[ "^/dev/mapper/" ]
+    ignore_suspended_devices=1
+    write_cache_state=0
+    disable_after_error_count=3
+    filter=[ "a|^/dev/mapper/.*|", "r|.*|" ]
+}
+"""
+LVM_CONFIG = re.sub(r"\s+", " ", LVM_CONFIG).strip()
+
+
+class Vdsm(Plugin, RedHatPlugin):
+    """vdsm Plugin"""
+
+    packages = (
+        'vdsm',
+        'vdsm-client',
+    )
+
+    plugin_name = 'vdsm'
+
+    def setup(self):
+        os.environ["LVM_SYSTEM_DIR"] = "/var/run/vdsm/lvm"
+        self.add_forbidden_path('/etc/pki/vdsm/keys/*')
+        self.add_forbidden_path('/etc/pki/vdsm/libvirt-spice/*-key.*')
+        self.add_forbidden_path('/etc/pki/libvirt/private/*')
+
+        self.add_cmd_output('service vdsmd status')
+
+        self.add_copy_spec(['/tmp/vds_installer*',
+                            '/tmp/vds_bootstrap*',
+                            '/etc/vdsm/*'])
+
+        limit = self.get_option('log_size')
+
+        if limit is not None:
+            self.add_copy_spec('/var/log/vdsm/*', sizelimit=limit)
+        else:
+            self.add_copy_spec('/var/log/vdsm/*')
+
+        self._add_vdsm_run_dir()
+        self.add_copy_spec(['/usr/libexec/vdsm/hooks',
+                            '/var/lib/vdsm',
+                            '/var/log/ovirt.log',
+                            '/var/log/sanlock.log'])
+
+        res = self.call_ext_prog('/usr/bin/pgrep qemu-kvm')
+        if res['status'] == 0:
+            for line in res['output']:
+                pid = line.strip()
+                self.add_copy_spec(["/proc/%s/cmdline" % pid,
+                                    "/proc/%s/status" % pid,
+                                    "/proc/%s/mountstats" % pid])
+        self.add_cmd_output(["/bin/ls -ldZ /etc/vdsm",
+                            "/bin/su vdsm -s /bin/sh -c\
+                             '/usr/bin/tree -l /rhev/data-center'",
+                             "/bin/su vdsm -s /bin/sh -c\
+                              '/bin/ls -lR /rhev/data-center'"])
+        self.add_cmd_output("/sbin/lvm vgs -v -o +tags --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_vgs_v_o_tags")
+        self.add_cmd_output("/sbin/lvm lvs -v -o +tags --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_lvs_v_o_tags")
+        self.add_cmd_output("/sbin/lvm pvs -v -o +all --config \'%s\'"
+                            % LVM_CONFIG,
+                            suggest_filename="sbin_lvm_pvs_v_o_all")
+        self.add_cmd_output(["/sbin/fdisk -l",
+                             "/usr/bin/iostat",
+                             "/sbin/iscsiadm -m node",
+                             "/sbin/iscsiadm -m session",
+                             "/usr/sbin/nodectl info",
+                             "/usr/bin/abrt-cli list"])
```

Comment:
The `abrt-cli list` is already collected by `abrt` plugin - is there a reason to collect it also here?

Reply:
can we add mandatory requirement for sos abrt plugin with this plugin? if yes I will omit it from the output

Reply:
Nack to all of this. These are collected in other plugins and this is not the approach we use in sos: each plugin gathers only the information relevant to the component it supports. This is the only way to make things manageable with hundreds of plugins and not end up in a mess of duplicate and confusing data placement.

We can possibly make an exception for the LVM data for `vdsm`, since it uses it in a very particular way, and we may not be able to immediately find a clean way to integrate this into the `lvm` plugin, but for these generic commands it's a very strong nack.